python/python/lancedb/query.py

```diff
@@ -544,6 +550,30 @@ def where(self, where: str, prefilter: bool = False) -> LanceVectorQueryBuilder:
         self._prefilter = prefilter
         return self
 
+    def rerank(self, reranker: Reranker, query: str) -> LanceVectorQueryBuilder:
+        """Rerank the results using the specified reranker.
+
+        Parameters
+        ----------
+        reranker: Reranker
+            The reranker to use.
+
+        query: str
+            The query to use for reranking. This needs to be specified explicitly here
+            as the query used for vector search may already be vectorized and the
+            reranker requires a string query.
+            Note: This doesn't yet support the case where the query is multimodal or a
+            list of vectors.
+
+        Returns
+        -------
+        LanceVectorQueryBuilder
+            The LanceQueryBuilder object.
+        """
+        self._reranker = reranker
+        self._reranker_query = query
```

Comment:
we can add logic for adding string query only if original query was a vector search